### PR TITLE
[ENG-2201] Registries Moderation Notifications

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Add notifications to registries moderation machine

## Changes

- Makes the preprint moderation email templates generic to registrations.
- Makes Preprint provider moderation notifications a sub-class of both  PreprintProvider and RegistrationProvider
- Makes adding a moderator to a provider add notification subscriptions for withdraw requests
- Adds many tests.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify emails are sent at appropriate times
- Verify emails are grammatically correct
- Verify no additional error messages appear during notification actions.

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Documentation

No new docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2201

### CODE REVIEW

File: osf/migrations/0219_add_registration_provider_subscription.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import logging
```
Comment (Johnetordoff) [inline osf/migrations/0219_add_registration_provider_subscription.py:1]: merge with 220

File: osf/migrations/__init__.py [hunk 2]
```diff
@@ -125,3 +125,60 @@ def update_permission_groups(sender, verbosity=0, **kwargs):
     if getattr(sender, 'label', None) == 'osf':
         update_admin_permissions(verbosity)
         update_provider_auth_groups(verbosity)
+
```
Comment (Johnetordoff) [inline osf/migrations/__init__.py:128]: Make sure not a dupe

File: osf/migrations/__init__.py [hunk 3]
```diff
@@ -125,3 +125,60 @@ def update_permission_groups(sender, verbosity=0, **kwargs):
...
+        for group_name, group_permissions in ReviewProviderMixin.groups.items():
+            group, created = Group.objects.get_or_create(name=self.format_group(group_name))
+            to_remove = set(get_perms(group, self.provider)).difference(group_permissions)
+            for p in to_remove:
+                remove_perm(p, group, self.provider)
+            for p in group_permissions:
+                assign_perm(p, group, self.provider)
+
+
+def populate_provider_notification_subscriptions_registration(app=None, state=None):
```
Comment (Johnetordoff) [inline osf/migrations/__init__.py:163]: make post migration signal, remove preprintprovider legacy migration and tinker more

File: osf/models/mixins.py [hunk 4]
```diff
@@ -2325,9 +2337,9 @@ def run_accept(self, user, comment, embargo_end_date=None):
             comment: Text describing why.
         """
         if embargo_end_date:
-            return self._run_transition(self.TriggersClass.EMBARGO.value, user=user, end_date=embargo_end_date)
+            return self._run_transition(self.TriggersClass.EMBARGO.value, user=user, comment=comment, end_date=embargo_end_date)
```
Comment (Johnetordoff) [inline osf/models/mixins.py:2340]: oops

File: osf/models/provider.py [hunk 5]
```diff
@@ -327,15 +331,18 @@ def rules_to_subjects(rules):
 
 
 @receiver(post_save, sender=PreprintProvider)
+@receiver(post_save, sender=RegistrationProvider)
 def create_provider_auth_groups(sender, instance, created, **kwargs):
     if created:
         instance.update_group_permissions()
+        update_provider_auth_groups()
```
Comment (Johnetordoff) [inline osf/models/provider.py:338]: look at more/remove

File: osf/utils/machines.py [hunk 6]
```diff
@@ -341,14 +358,23 @@ def get_context(self):
...
         }
 
 
-class RegistrationMachine(BaseMachine):
+class RegistrationMachine(ModerationNotificationMixin):
     ActionClass = RegistrationAction
     States = RegistrationStates
     Transitions = REGISTRATION_TRANSITIONS
 
+    def get_request_action(self):
```
Comment (Johnetordoff) [inline osf/utils/machines.py:370]: Probably unnecesarry

File: osf/utils/machines.py [hunk 7]
```diff
@@ -385,7 +411,7 @@ def embargo_registration(self, ev):
         user = ev.kwargs.get('user')
         end_date = ev.kwargs.get('end_date')
         self.machineable.refresh_from_db()
-        self.machineable.registered_node.embargo_registration(user, end_date)
+        self.machineable.registered_node.embargo_registration(user, end_date, notify_initiator_on_complete=True)
```
Comment (Johnetordoff) [inline osf/utils/machines.py:414]: check what `notify_initiator_on_complete` does

File: osf/utils/machines.py [hunk 8]
```diff
@@ -430,27 +456,72 @@ def withdrawal_request_fails(self, ev):
     def force_withdrawal(self, ev):
         pass
 
-    def notify_embargo_termination(self, ev):
-        pass
+    def notify_request_withdraw(self, ev):
```
Comment (Johnetordoff) [inline osf/utils/machines.py:459]: RequestWithdraw mixin?

File: osf/utils/machines.py [hunk 9]
```diff
@@ -430,27 +456,72 @@ def withdrawal_request_fails(self, ev):
...
-    def notify_embargo(self, ev):
-        pass
+    def notify_withdraw_registration(self, ev):
+        context = self.get_context()
+        for contributor in self.machineable.registered_node.contributors.all():
+            context['contributor'] = contributor
+            if context.get('requester', None):
+                context['is_requester'] = context['requester'].username == contributor.username
 
-    def notify_withdraw(self, ev):
```
Comment (Johnetordoff) [inline osf/utils/machines.py:439]: Too indented

File: osf/utils/machines.py [hunk 10]
```diff
@@ -430,27 +456,72 @@ def withdrawal_request_fails(self, ev):
...
+            context['withdrawal_submitter_is_moderator_or_admin'] = True
 
-    def notify_edit_comment(self, ev):
-        pass
+        for contributor in self.machineable.registered_node.contributors.all():
+            context['contributor'] = contributor
+            if context.get('requester', None):
+                context['is_requester'] = context['requester'].username == contributor.username
 
-    def notify_resubmit(self, ev):
```
Comment (Johnetordoff) [inline osf/utils/machines.py:448]: too indented

File: osf_tests/test_registration_moderation_notifications.py [hunk 11]
```diff
@@ -0,0 +1,475 @@
+import pytest
```
Comment (Johnetordoff) [inline osf_tests/test_registration_moderation_notifications.py:1]: Split moderation and withdraw actions up

File: website/reviews/listeners.py [hunk 12]
```diff
@@ -112,7 +188,7 @@ def reviews_withdrawal_requests_notification(self, timestamp, context):
                                                                            preprint.provider._id,
                                                                            preprint._id)
     # Store emails to be sent to subscribers instantly (at a 5 min interval)
-    emails.store_emails(provider_subscription.email_transactional.all().values_list('guids___id', flat=True),
+    store_emails(provider_subscription.email_transactional.all().values_list('guids___id', flat=True),
```
Comment (Johnetordoff) [inline website/reviews/listeners.py:191]: fix these and make sure they are tested

File: website/templates/emails/new_pending_withdraw_requests.html.mako [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<table class="comment-row" border="0" cellpadding="8" cellspacing="0" width="100%" align="center" style="font-size: 13px;background: #fff;border: 1px solid #eee;border-radius: 5px;margin-bottom: 10px;padding: 0px !important;">
```
Comment (Johnetordoff) [inline website/templates/emails/new_pending_withdraw_requests.html.mako:1]: Read this and get Product language

File: osf/migrations/__init__.py [hunk 14]
```diff
@@ -125,3 +125,30 @@ def update_permission_groups(sender, verbosity=0, **kwargs):
     if getattr(sender, 'label', None) == 'osf':
         update_admin_permissions(verbosity)
         update_provider_auth_groups(verbosity)
+
+
```
Comment (corbinSanders) [inline osf/migrations/__init__.py:129]: Drive by: Will this overwrite any existing preprint moderator notification settings?

**Review by UdayVarkhedkar — APPROVED** [review]
Outside of rebasing/merging with `develop`, this looks like a really good and very thorough job! 

edited: Removed note about modifying migration numbering after acknowledging that 0219 here should modify 0219 in develop.

File: osf/migrations/__init__.py (hunk 14 above)
Comment (Johnetordoff) [reply osf/migrations/__init__.py:129]: No, there's a check in `add_user_to_subscription` that makes it a noop if the person is already subscribed.

File: website/templates/emails/new_pending_withdraw_requests.html.mako (hunk 13 above)
Comment (brianjgeiger) [reply website/templates/emails/new_pending_withdraw_requests.html.mako:1]: Do we still need something here?

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Migration cleanup still needs to be done properly. Plus a few questions.

File: osf/migrations/__init__.py [hunk 15]
```diff
@@ -125,3 +125,30 @@ def update_permission_groups(sender, verbosity=0, **kwargs):
...
+def populate_provider_subscriptions(app=None, state=None):
+    from django.apps import apps
+
+    update_provider_auth_groups()  # for test env
+
+    NotificationSubscription = apps.get_model('osf', 'NotificationSubscription')
+    RegistrationProvider = apps.get_model('osf', 'RegistrationProvider')
+    PreprintProvider = apps.get_model('osf', 'PreprintProvider')
+
+    for provider_class in [RegistrationProvider, PreprintProvider]:
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/migrations/__init__.py]: 1. Users who have removed themselves are excluded [here](https://github.com/CenterForOpenScience/osf.io/pull/9496/files#diff-cdd3245c8425595695344c6f48ae4e41b5af1b457ec70ecd9e4fdab132e7714dR1024-R1031) 

2. Yeah that makes no sense, but I'm reworking it as a post-migration signal and hopefully that will automate the process of adding provider subscriptions.

File: osf/models/mixins.py (hunk 4 above)
Comment (Johnetordoff) [reply osf/models/mixins.py:2339]: it's required in the email templates, so it's necessary

File: osf/migrations/__init__.py (hunk 14 above)
Comment (Johnetordoff) [reply osf/migrations/__init__.py:129]: If they did, then they should be in the none relationship of the notification

File: osf/utils/machines.py [hunk 16]
```diff
@@ -430,32 +447,75 @@ def withdrawal_request_fails(self, ev):
...
+            )
 
-    def notify_accept_reject(self, ev):
-        pass
+    def notify_reject_withdraw_request(self, ev):
+        context = self.get_context()
+        try:
+            action = RegistrationAction.objects.get(
+                target__id=self.machineable.registered_node.id,
+                from_state='pending',
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/utils/machines.py:473]: @jwalz any developments on this?

File: osf/utils/machines.py [hunk 17]
```diff
@@ -97,7 +98,66 @@ def update_last_transitioned(self, ev):
         self.machineable.date_last_transitioned = now
 
 
-class ReviewsMachine(BaseMachine):
+class ModerationNotificationMixin(BaseMachine):
```
Comment (jwalz) [inline osf/utils/machines.py:101]: Driveby: This really doesn't need to be a Machine subclass. All you're gaining is direct access to "machineable" and saving a step by not having to write your state machine callbacks.

In general, I'd say this is cleaner as a helper class (or just util module?) that generates and sends emails given a registration/preprint and trigger

File: osf/utils/machines.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (jwalz) [reply osf/utils/machines.py:473]: A moderator-rejected withdrawal would go from 'pending_withdraw' to either 'accepted' or 'embargo' depending on the state of the underlying Embargo/RegistrationApproval sanction.

You could just get based on the REJECT_WITHDRAWAL trigger?

File: website/templates/emails/new_pending_withdraw_requests.html.mako (hunk 13 above)
Comment (Johnetordoff) [reply website/templates/emails/new_pending_withdraw_requests.html.mako:1]: So this was to add digests for withdraw requests but that's not in the reqs so I'm deleting it.

File: osf/migrations/0224_population_registration_subscription_notifications.py [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from osf.migrations import populate_registration_provider_subscriptions
+
+
+def revert(apps, schema_editor):
+    NotificationSubscription = apps.get_model('osf', 'NotificationSubscription')
+    # The revert of this migration deletes all NotificationSubscription instances
```
Comment (brianjgeiger) [inline osf/migrations/0224_population_registration_subscription_notifications.py:12]: Including existing preprint moderator notifications? That seems a bit broad.

File: osf/models/registrations.py [hunk 19]
```diff
@@ -407,7 +422,7 @@ def terminate_embargo(self, forced=False):
         if not self.is_embargoed:
             raise NodeStateError('This node is not under active embargo')
 
-        action = NodeLog.EMBARGO_TERMINATED if forced else NodeLog.EMBARGO_COMPLETED
+        action = NodeLog.EMBARGO_COMPLETED if not forced else NodeLog.EMBARGO_TERMINATED
```
Comment (brianjgeiger) [inline osf/models/registrations.py:425]: This should be changed back

```suggestion
        action = NodeLog.EMBARGO_TERMINATED if forced else NodeLog.EMBARGO_COMPLETED
```

File: osf_tests/test_registration_moderation_notifications.py [hunk 20]
```diff
@@ -0,0 +1,359 @@
...
+            trigger=RegistrationModerationTriggers.REQUEST_WITHDRAWAL.value,
+            from_state=RegistrationModerationStates.ACCEPTED.value,
+            to_state=RegistrationModerationStates.PENDING_WITHDRAW.value,
+            comment='yo'
+        )
+        return registration_action
+
+    def test_submit_notifications(self, registration, moderator, admin, contrib, provider):
+        """
+        "As moderator of branded registry, I receive email notification upon admin author(s) submission approval"
```
Comment (brianjgeiger) [inline osf_tests/test_registration_moderation_notifications.py:90]: These are great! They would be even better if they included the REQS ticket number.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Drive-by: A couple of small comments and a question.

File: osf/migrations/0224_population_registration_subscription_notifications.py [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0223_auto_20201026_1843'),
+    ]
+
+    operations = [
+        migrations.RunPython(populate_registration_provider_subscriptions, revert)
```
Comment (UdayVarkhedkar) [inline osf/migrations/0224_population_registration_subscription_notifications.py:23]: Is this necessary given that there's a `populate_registration_provider_notification_subscriptions.py` management command that we could run after deployment? Or rather, does this make the management command unnecessary?

File: website/reviews/listeners.py [hunk 22]
```diff
@@ -28,11 +31,18 @@ def reviews_notification(self, creator, template, context, action):
 def reviews_submit_notification(self, recipients, context):
     # Avoid AppRegistryNotReady error
     from website.notifications.emails import get_user_subscriptions
-    from website import settings
+
+    PreprintProvider = apps.get_model('osf.PreprintProvider')
+    RegistrationProvider = apps.get_model('osf.RegistrationProvider')
 
     event_type = utils.find_subscription_type('global_reviews')
     if context['reviewable'].provider._id == 'osf':
```
Comment (jwalz) [inline website/reviews/listeners.py:39]: trust the document_type in the context as an invariant instead of explicitly checking the type of the provider?

File: osf/utils/notifications.py [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+        template='reviews_resubmission_confirmation',
+        action=action
+    )
+
+
+def notify_accept_reject(resource, user, action, states, *args, **kwargs):
+    context = get_email_template_context(resource)
+
+    context['notify_comment'] = not resource.provider.reviews_comments_private and action.comment
+    context['comment'] = action.comment
```
Comment (jwalz) [inline osf/utils/notifications.py:50]: It's fine fo the context to have unread values, right? Can we shove this into `get_email_template_context` so the comment is always available if we want it?

File: osf/utils/notifications.py [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+        template='reviews_submission_status',
+        action=action
+    )
+
+
+def notify_edit_comment(resource, user, action, *args, **kwargs):
+    context = get_email_template_context(resource)
+
+    context['comment'] = action.comment
+    if not resource.provider.reviews_comments_private and action.comment:
```
Comment (jwalz) [inline osf/utils/notifications.py:66]: nit: I'd do this check at the start and return early if it's False

**Review by jwalz — CHANGES_REQUESTED** [review]
Looking good, just a few nits, suggestions, and questions (and one likely bug)

File: osf_tests/test_registration_moderation_notifications.py [hunk 25]
```diff
@@ -0,0 +1,359 @@
...
+        :return:
+        """
+
+        with mock.patch('website.reviews.listeners.mails.send_mail') as mock_send_mail:
+            notify_submit(registration, admin)
+
+        assert len(mock_send_mail.call_args_list) == 2
+        admin_message, contrib_message = mock_send_mail.call_args_list
+
+        assert admin_message == call(
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf_tests/test_registration_moderation_notifications.py:102]: I don't consider these change detectors so much as integration tests, consider that email is a service and a wrong email cannot be unsent, but I will see if I can pick out some of the more boilerplate stuff.

File: osf/utils/notifications.py [hunk 26]
```diff
@@ -0,0 +1,98 @@
+from django.utils import timezone
+from website.mails import mails
+from website.reviews import signals as reviews_signals
+from website.settings import DOMAIN, OSF_SUPPORT_EMAIL, OSF_CONTACT_EMAIL
+
+
+def get_email_template_context(resource):
+    from osf.models import Preprint, Registration
+    assert isinstance(resource, (Registration, Preprint)), 'This resource does not fit the template'
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/utils/notifications.py:9]: Evaluating these at the top of the module creates circular imports. And yeah we could just check `.__class__.__name__` I believe.

File: osf/utils/notifications.py [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+    )
+    reviews_signals.reviews_email_submit_moderators_notifications.send(
+        timestamp=timezone.now(),
+        context=context
+    )
+
+
+def notify_resubmit(resource, user, action, *args, **kwargs):
+    context = get_email_template_context(resource)
+    reviews_signals.reviews_email.send(
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/utils/notifications.py:38]: I believe these are signals so the email queuing doesn't slow down the rest of the request, but obviously I didn't benchmark or anything.

File: osf/utils/notifications.py (hunk 23 above)
Comment (Johnetordoff) [reply osf/utils/notifications.py:50]: I would prefer if `get_email_template_context` is just derived from the resource, the comment is coming from the action, which it not always being passed by the Preprint machine.

File: website/reviews/listeners.py (hunk 22 above)
Comment (Johnetordoff) [reply website/reviews/listeners.py:39]: `document_type` could be any string, for preprints it can be `preprint_word` which is a custom word our provider picks, like "paper" or "publication" instead of "preprint".

File: osf/utils/notifications.py [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+        creator=user,
+        context=context,
+        template='reviews_resubmission_confirmation',
+        action=action
+    )
+
+
+def notify_accept_reject(resource, user, action, states, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/utils/notifications.py:46]: It would be somewhat significant we'd have to break up the digests and transnational emails, it's unclear how long that would take as we re-using a lot of preprint machine code.

[the remark this answers is not in the document]
Comment (jwalz) [reply osf/utils/notifications.py:46]: sounds good

File: osf/utils/notifications.py (hunk 23 above)
Comment (jwalz) [reply osf/utils/notifications.py:50]: sounds good

File: osf/utils/notifications.py (hunk 27 above)
[the remark this answers is not in the document]
Comment (jwalz) [reply osf/utils/notifications.py:38]: cool

File: website/reviews/listeners.py (hunk 22 above)
Comment (Johnetordoff) [reply website/reviews/listeners.py:39]: I don't really want to check the document_type, it's better to just check the provider type.

File: osf/utils/notifications.py (hunk 26 above)
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/utils/notifications.py]: now just checks the model type

File: osf/utils/machines.py [hunk 29]
```diff
@@ -147,33 +150,15 @@ def notify_submit(self, ev):
...
     def notify_edit_comment(self, ev):
-        context = self.get_context()
-        context['comment'] = self.action.comment
-        if not self.machineable.provider.reviews_comments_private and self.action.comment:
-            reviews_signals.reviews_email.send(creator=ev.kwargs.get('user'), context=context,
-                                               template='reviews_update_comment',
-                                               action=self.action)
+        notify_edit_comment(self.machineable, ev.kwargs.get('user'), self.action)
 
     def notify_withdraw(self, ev):
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/utils/machines.py:152]: Ah so I just thought this was contained in the Sanction notifications, but it's present now.

**Review by jwalz — CHANGES_REQUESTED** [review]

File: osf/utils/notifications.py [hunk 30]
```diff
@@ -5,10 +5,6 @@
 from osf.utils.workflows import RegistrationModerationTriggers
 
 def get_email_template_context(resource):
-    provider = getattr(resource, 'provider', False)
-    assert provider, 'This resource does not fit the template'
```
Comment (jwalz) [inline osf/utils/notifications.py]: Is it possible/likely that this will be called with resources that don't have a provider? If so, we want to do these checks instead of letting things die with an AttributeError, we should just use more expressive exceptions than `assert` outside of test code. If it's not likely/currently possible, then this is fine.

File: osf/utils/notifications.py [hunk 31]
```diff
@@ -5,10 +5,6 @@
...
-    assert provider.type in ['osf.preprintprovider', 'osf.registrationprovider']
-
     is_preprint = resource.provider.type == 'osf.preprintprovider'
```
Comment (jwalz) [inline osf/utils/notifications.py:8]: silly question, can we check resource.type instead of resource.provider.type?

File: osf/utils/notifications.py [hunk 32]
```diff
@@ -101,32 +97,14 @@ def notify_moderator_registration_requests_withdrawal(resource, user, *args, **k
...
-    withdrawal_requester = resource.actions.filter(
-        trigger=RegistrationModerationTriggers.REQUEST_WITHDRAWAL.db_name,
-        is_deleted=False
-    ).get().creator
+    context['force_withdrawal'] = action.trigger == RegistrationModerationTriggers.FORCE_WITHDRAW.db_name
 
     for contributor in resource.contributors.all():
         context['contributor'] = contributor
-        context['requester'] = withdrawal_requester
+        context['requester'] = resource.retraction.initiated_by
```
Comment (jwalz) [inline osf/utils/notifications.py:107]: nit: this can be done outside the for loop

File: website/reviews/listeners.py [hunk 33]
```diff
@@ -86,7 +86,7 @@ def reviews_submit_notification_moderators(self, timestamp, context):
     elif provider.type == 'osf.registrationprovider':
         url_segment = 'registries'
     else:
-        raise NotImplementedError()
+        raise NotImplementedError('unsupported provider type')
```
Comment (jwalz) [inline website/reviews/listeners.py:89]: include the provider type in the message, please

**Review by jwalz — APPROVED** [review]
Just minor nits remaining, but not 100% confident on the migration stuffs, so asking for one more set of eyes

File: osf/migrations/0224_population_registration_subscription_notifications.py [hunk 34]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def revert(apps, schema_editor):
+    NotificationSubscription = apps.get_model('osf', 'NotificationSubscription')
+    # The revert of this migration deletes all NotificationSubscription instances
+    NotificationSubscription.objects.filter(provider__isnull=False, provider__type='osf.registrationprovider').delete()
+
+
+def populate_subscriptions(*args, **kwargs):
+    add_reviews_notification_setting('global_reviews')
```
Comment (brianjgeiger) [inline osf/migrations/0224_population_registration_subscription_notifications.py:18]: Doesn't this get added at the beginning of `populate_registration_provider_notification_subscriptions()`?

File: osf/utils/notifications.py (hunk 31 above)
Comment (Johnetordoff) [reply osf/utils/notifications.py:8]: No, type here is only reffering to typed models, preprint used to be one, but is now it's own thing, separate from nodes.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]

File: osf/migrations/0224_population_registration_subscription_notifications.py (hunk 34 above)
Comment (Johnetordoff) [reply osf/migrations/0224_population_registration_subscription_notifications.py:18]: ah yep, it's a dupe

File: osf/management/commands/populate_registration_provider_notification_subscriptions.py [hunk 35]
```diff
@@ -26,8 +26,6 @@ def populate_registration_provider_notification_subscriptions():
                 event_name=subscription,
                 provider=provider
             )
-            if created:
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/management/commands/populate_registration_provider_notification_subscriptions.py]: nope

**Review by brianjgeiger — APPROVED** [review]

**Comment by ajski** [comment]
@Johnetordoff if you have time, might be cool to take care of this one: https://openscience.atlassian.net/browse/ENG-2401 (or, can be deferred)
